### PR TITLE
Convert `:not` pseudo-class selector list to chained

### PR DESCRIPTION
**Pull request by aaronccasanova**
Continuation of [PR# 9556](https://github.com/Shopify/polaris/pull/9556) applying the quick fix to all `:not` psuedo-classes with a selector list.

Non-exhaustive pattern to find `:not` psuedo-classes with a selector list:
image

| Before | After |
|--------|--------|
| Screenshot 2023-06-28 at 2 23 07 PM | Screenshot 2023-06-28 at 2 26 13 PM |

| Before | After |
|--------|--------|
| Screenshot 2023-06-28 at 2 32 06 PM | Screenshot 2023-06-28 at 2 30 37 PM |

| Before | After |
|--------|--------|
| Screenshot 2023-06-28 at 2 34 29 PM | Screenshot 2023-06-28 at 2 36 17 PM |

### CODE REVIEW

**Review by sophschneider — APPROVED** [review]

File: polaris-react/src/components/ResourceItem/ResourceItem.scss [hunk 1]
```diff
@@ -128,11 +128,13 @@
     }
   }
 
-  &:not(.hasBulkActions, .selectMode) {
```
Comment (aaronccasanova) [inline polaris-react/src/components/ResourceItem/ResourceItem.scss:131]: Note: This PR updates a mainline `:not` psuedo-class (e.g. not beta-flagged) (Screenshot in PR description)

**Review by chloerice — APPROVED** [review]
Thank you so much for flagging and fixing this Aaron!!

**Review by sam-b-rose — APPROVED** [review]
I do`:not:not` approve this PR 👍 💯  nice work catching this

File: polaris-react/src/components/DropZone/DropZone.scss [hunk 2]
```diff
@@ -105,7 +105,7 @@ $dropzone-stacking-order: (
     }
   }
 
-  #{$se23} &:not(.focused, .isDisabled) {
+  #{$se23} &:not(.focused):not(.isDisabled) {
```
Comment (chloerice) [inline polaris-react/src/components/DropZone/DropZone.scss:108]: Tophat of the [ProductDetails](https://admin.web.fix-not-selector-lists.aaron-casanova.us.spin.dev/store/shop1/products/2) DropZone looks good 💯 

Screenshot 2023-06-29 at 4 52 35 PM

File: polaris-react/src/components/ResourceItem/ResourceItem.scss [hunk 3]
```diff
@@ -128,11 +128,13 @@
...
+  &:not(.hasBulkActions):not(.selectMode) {
```
Comment (chloerice) [inline polaris-react/src/components/ResourceItem/ResourceItem.scss:131]: Tophat of ProductDetails OptionsAndVariantsCard with multi-location [looks good](https://admin.web.fix-not-selector-lists.aaron-casanova.us.spin.dev/store/shop6/products/3) 👍🏽 

Screenshot 2023-06-29 at 6 47 24 PM

**Comment by aaronccasanova** [comment]
Thanks for the tophatting assistance @chloerice ❤️

File: polaris-react/src/components/Button/Button.scss [hunk 4]
```diff
@@ -911,7 +911,7 @@
     }
 
     #{$se23} &:hover,
-    #{$se23} &:focus:not(:active, .pressed, .disabled) {
+    #{$se23} &:focus:not(:active):not(.pressed):not(.disabled) {
```
Comment (chloerice) [inline polaris-react/src/components/Button/Button.scss:914]: Tophat of [AppIndexResourceListItem](https://admin.web.fix-not-selector-lists.aaron-casanova.us.spin.dev/store/shop6/settings/apps?tab=installed) looks good 🥇 

Screenshot 2023-06-29 at 6 57 18 PM

**Pull request merged**